src/NLog/LogFactory.cs

```diff
@@ -338,7 +338,7 @@ public LoggingConfiguration Configuration
                         }
                     }
 
-                    this.OnConfigurationChanged(new LoggingConfigurationChangedEventArgs(value, oldConfig));
+                    this.OnConfigurationChanged(new LoggingConfigurationChangedEventArgs(oldConfig, value));
```

Comment:
so this is still a breaking change. 

I like to resolve this issue, but if it's breaking, it should be NLog 5.